browser/mocha/stub.js

```diff
@@ -25,20 +33,17 @@ extendInterfaces('stub', function(context, teardown) {
     var proto = document.createElement(tagName).constructor.prototype;
 
     // For all keys in the implementation to stub with..
-    var keys = Object.keys(implementation);
-    keys.forEach(function(key) {
+    var stubs = Object.keys(implementation).map(function(key) {
       // Stub the method on the element prototype with Sinon:
-      sinon.stub(proto, key, implementation[key]);
+      return sinon.stub(proto, key, implementation[key]);
     });
 
     // After all tests..
     teardown(function() {
       // For all of the keys in the implementation we stubbed..
-      keys.forEach(function(key) {
+      stubs.forEach(function(stub) {
```

Comment:
The logic here is roughly the same, except for remembering and operating on stubs directly. Can you explain why this was failing before?